backend/app/notify/notify.go

```diff
@@ -29,14 +29,27 @@ type Destination interface {
 	Send(ctx context.Context, req request) error
 }
 
-// Store defines the minimal interface accessing stored comments used by notifier
+// VerificationDestination defines interface for a given destination service which supports verification sending
+type VerificationDestination interface {
+	SendVerification(ctx context.Context, req VerificationRequest) error
+}
+
+type VerificationRequest struct {
+	Locator store.Locator
+	User    string
+	Email   string
+	Token   string
+}
+
+// Store defines the minimal interface accessing stored comments and retrieving users details used by notifier
 type Store interface {
 	Get(locator store.Locator, id string, user store.User) (store.Comment, error)
 }
 
 type request struct {
-	comment store.Comment
-	parent  store.Comment
+	comment         store.Comment
+	parent          store.Comment
+	parentUserEmail string
```

Comment:
this one is too specific for notify and it is leaking abstraction. Notify suppose to work on more generic level of abstaction - get a pair of comments ane maybe some meta and do the job. In no place it should be dealing with email directly

Reply:
The problem is that whoever notify. Submit calls does not have any access to datastore but should have all context for sending message already.

Another solution I can see is having CommentWitheEmail structure for sending via notify. Submit, and asking the sender to fill an email for it. Would it be better?

Reply:
I meant no need to make it email-related so directly. Just add some meta struct holding all info we need in addition to the comment. i.e. Submit(req Request) and Request may include the comment, email and whatever else we need in the future